runtime-light/stdlib/zlib/zlib-functions.cpp

```diff
@@ -115,3 +131,152 @@ std::optional<string> decode(std::span<const char> data, int64_t encoding) noexc
 }
 
 } // namespace kphp::zlib
+
+class_instance<C$DeflateContext> f$deflate_init(int64_t encoding, const array<mixed>& options) noexcept {
+  int32_t level{-1};
+  int32_t memory{8};
+  int32_t window{15};
+  auto strategy{Z_DEFAULT_STRATEGY};
+  auto extract_int_option{[](int32_t lbound, int32_t ubound, const array_iterator<const mixed>& option, int32_t& dst) noexcept {
```

Comment:
Let's mark is as `constexpr`

Reply:
> Like this (see changes)? What does it do?

This allows the value of a variable to be calculated at compile time
https://en.cppreference.com/w/cpp/language/constexpr.html